### ISSUE TITLE
Size mismatch error when merge LoRA weights

### ISSUE THREAD

**Issue by JoeLeelyf** [haotian-liu/LLaVA#462]
### Describe the issue

Issue: I was trying to do down-stream fine-tune training on LLaVA with the LLM "llama-2-7b-chat". Since that only LoRA model for the 7b-chat version was published in the model zoo, I made my attempt to merge the LoRA and the original llama-2-7b model using the following command, but met with a size mismatch error.

Command:
```
python merge_lora_weights.py --model-path liuhaotian/llava-llama-2-7b-chat-lightning-lora-preview --model-base meta-llama/Llama-2-7b-chat-hf --save-model-path ./checkpoints/llava-7b-llama-2-7b-chat
```

Log: 
```
Traceback (most recent call last):
  File "/mnt/data-2/data/lyf/LLaVA/scripts/merge_lora_weights.py", line 22, in <module>
    merge_lora(args)
  File "/mnt/data-2/data/lyf/LLaVA/scripts/merge_lora_weights.py", line 8, in merge_lora
    tokenizer, model, image_processor, context_len = load_pretrained_model(args.model_path, args.model_base, model_name, device_map='cpu')
  File "/mnt/data-2/data/yh/LLaVA_3D/llava/model/builder.py", line 69, in load_pretrained_model
    model.load_state_dict(non_lora_trainables, strict=False)
  File "/mnt/data-1/data/jiagang.zhu/miniconda3/envs/llava/lib/python3.10/site-packages/torch/nn/modules/module.py", line 2041, in load_state_dict
    raise RuntimeError('Error(s) in loading state_dict for {}:\n\t{}'.format(
RuntimeError: Error(s) in loading state_dict for LlavaLlamaForCausalLM:
        size mismatch for model.mm_projector.weight: copying a param with shape torch.Size([4096, 1024]) from checkpoint, the shape in current model is torch.Size([4096, 20]).
```

I see that others have successfully made this in #410, and I don't know why mine is not working. The LLaVA codes are the latest version which I cloned to my computer just now.

**Comment by makanju0la** [comment]
Was your pre-trained model trained on Vicuna 7b ? I had this same error and it was because I was trying to merge my 7b parameter Lora model with the 13b base model like this:

> python3 scripts/merge_lora_weights.py --model-path /home/grads/mogunleye/Research/LLaVA/checkpoints/llava-v1.5-7b-lora --model-base **liuhaotian/llava-v1.5-13b** --save-model-path /home/grads/mogunleye/Research/LLaVA/checkpoints/llava-merged_lora

When I fixed this and used the corresponding base model that aligns with my fine-tuned model, like below, I was able to merge successfully: 

> python3 scripts/merge_lora_weights.py --model-path /home/grads/mogunleye/Research/LLaVA/checkpoints/llava-v1.5-7b-lora --model-base **liuhaotian/llava-v1.5-7b** --save-model-path /home/grads/mogunleye/Research/LLaVA/checkpoints/llava-merged_lora 

TLDR: use the correct base model might help. I was using 13B LLaVa for a 7b  trained Lora
